src/Microsoft.SqlTools.ServiceLayer/Utility/SqlScriptFormatter.cs

```diff
@@ -176,6 +176,83 @@ public static string UnwrapLiteral(string literal)
             return literal;
         }
 
+        public static string[] DecodeMultipartIdenfitier(string multipartIdentifier)
+        {
+            StringBuilder sb = new StringBuilder();
+            List<string> namedParts = new List<string>();
+            bool insideBrackets = false;
+            bool bracketsClosed = false;
+            for (int i = 0; i < multipartIdentifier.Length; i++)
+            {
+                char iChar = multipartIdentifier[i];
+                if (insideBrackets)
+                {
+                    if (iChar == ']')
```

Comment:
No need to escape `[` when used inside square brackets. The syntax you listed only matters for values where quasi-regex patterns are being written (ie, in a `LIKE` clause).